internal/kgateway/extensions2/plugins/inferenceextension/endpointpicker/status.go

```diff
@@ -417,6 +412,67 @@ func updatePoolStatus(
 	}
 }
 
+// addGatewaysFromParentRef resolves a ParentReference on an HTTPRoute into concrete Gateway
+// names and adds them to the provided set. It supports:
+// - Kind: Gateway (group gateway.networking.k8s.io)
+// - Kind: XListenerSet (group gateway.networking.x-k8s.io)
+func addGatewaysFromParentRef(
+	ctx context.Context,
+	commonCol *common.CommonCollections,
+	routeNamespace string,
+	pr gwv1.ParentReference,
+	dest map[types.NamespacedName]struct{},
+) {
+	// Determine kind
+	kind := wellknown.GatewayKind
+	if pr.Kind != nil && *pr.Kind != "" {
+		kind = string(*pr.Kind)
+	}
+
+	// Determine group
+	group := gwv1.GroupName
+	if pr.Group != nil && *pr.Group != "" {
+		group = string(*pr.Group)
+	}
+
+	// Determine namespace defaulting to the HTTPRoute's namespace
+	ns := routeNamespace
+	if pr.Namespace != nil && *pr.Namespace != "" {
+		ns = string(*pr.Namespace)
+	}
+
+	switch kind {
+	case wellknown.GatewayKind:
+		if group != gwv1.GroupName {
+			return
+		}
+		dest[types.NamespacedName{Namespace: ns, Name: string(pr.Name)}] = struct{}{}
+		return
+	case wellknown.XListenerSetKind, wellknown.ListenerSetKind:
+		// Support experimental "XListenerSet" kind
+		if group != gwv1.GroupName && group != wellknown.XListenerSetGroup {
```

Comment:
```suggestion
		if group != wellknown.XListenerSetGroup {
```
XListenerSetGroup have their own group, so why are we comparing it against gwv1.GroupName ?